src/realm/util/safe_int_ops.hpp

```diff
@@ -19,6 +19,11 @@
 #ifndef REALM_UTIL_SAFE_INT_OPS_HPP
 #define REALM_UTIL_SAFE_INT_OPS_HPP
 
+#ifdef _WIN32
+#undef max // collides with numeric_limits::max called later in this header file
+#undef min // collides with numeric_limits::min called later in this header file
+#endif
+
```

Comment:
I'm trying to make it a bit more user friendly by removing the need for all the compiler command line flags and fixing it in the files instead. I need to use `undef` instead of NOMINMAX in this header because the std::min/max is defined through windows.h which the end user might have included *before* realm.hpp, so NOMINMAX would not be visible there